lib/src/Uncertainty/Model/DistributionImplementation.cxx

```diff
@@ -305,9 +294,14 @@ Distribution DistributionImplementation::operator * (const Scalar value) const
   if (dimension_ != 1) throw NotYetImplementedException(HERE) << "In DistributionImplementation::operator * (const Scalar value) const: can multiply by a constant 1D distributions only.";
   if (value == 0.0) return new Dirac(Point(1, 0.0));
   if (value == 1.0) return clone();
+  if (getClassName() == "Dirac") return new Dirac(getRealization()[0] * value);
   const Collection< Distribution > coll(1, *this);
   const Point weight(1, value);
-  return new RandomMixture(coll, weight);
```

Comment:
we might check `value` here and return `*this` in case it equals `1`

Reply:
I can't find this line in the code...

Reply:
its done just below that, note that its simpler to just check `value` instead of building the mixture:
```
if (value == 1.0)
  return *this;
```

Reply:
Please note that the special cases Distribution*0 and Distribution*1 are checked at the very beginning of the method (return clone() instead of *this, can be changed if needed). The *key point* is that even if value is not equal to 0 or 1, it can be simplified in the RandomMixture eg by being integrated into the mean/standard deviation of a Normal: 2*Normal(1,1)=Normal(2,2), which is done in the RandomMixture and is detected because there is a unique atom with a unit weight after the simplification process!

Reply:
ok, thanks